### PR TITLE
faster agreement sender

### PR DESCRIPTION
**Pull request by egonelbre**
Use streaming settlement, instead of immediately waiting for a response from satellite, do it asynchronously.

Storage Node keeps sending agreements.
Satellite resolves agreements as fast as it can.
Storage Node waits responses from satellite.

### CODE REVIEW

**Comment by mniewrzal** [comment]
Can you add a short description why it's faster?

File: pkg/bwagreement/server.go [hunk 1]
```diff
@@ -121,10 +124,91 @@ func (s *Server) BandwidthAgreements(ctx context.Context, rba *pb.Order) (reply
...
+		request, err := client.Recv()
+		if err != nil {
+			return formatError(err)
+		}
+
+		if request == nil || request.Allocation == nil {
+			return status.Error(codes.InvalidArgument, "allocation missing")
+		}
+		allocation := request.Allocation
+		payerAllocation := allocation.PayerAllocation
```
Comment (aleitner) [inline pkg/bwagreement/server.go]: Check if payerAllocation is nil

Comment (egonelbre) [reply pkg/bwagreement/server.go]: payerAllocation cannot be nil here

Comment (aleitner) [reply pkg/bwagreement/server.go]: I see now that it is a struct

**Comment by egonelbre** [comment]
Adding "Do not Merge", because waiting for testing results from Jens.

**Comment by littleskunk** [comment]
Tests are looking good. I manipulated my storj-sim setup. The sender interval for one storage node is set to 1 minute. I manipulated the timeout in the code to 1 minute as well and the satellite will not respond for 10 hours. With this PR the storage node agreement sender will not hang and keep sending.
```
satellite/0   1Z2GQsrLww 12:46:58.232 | DEBUG    agreements    Settlement sleep 10 hours    {"storage node ID": "12UzTUTgSe2hueY1W3Fd2V9ZUhMQp6yWMBfZBJfi2jec7QWYwtj"}
satellite/0   1Z2GQsrLww 12:46:58.239 | DEBUG    agreements    Settlement sleep 10 hours    {"storage node ID": "1ALvG9Lh1MePXNfVbCPsfdY2FEmoE2NrgxU96j16e4GJdaJRoY"}
satellite/0   1Z2GQsrLww 12:46:58.258 | DEBUG    agreements    Settlement sleep 10 hours    {"storage node ID": "1EVkWnG2uEDm6fDdLGMz1Mde2jqvHpkGre4CqjVv3tDAe26qMw"}
satellite/0   1Z2GQsrLww 12:46:58.267 | DEBUG    agreements    Settlement sleep 10 hours    {"storage node ID": "12usQccMrWnE1GsCXAxBVDm8m3sJumFpePt9oLtycVJj4VKD8Sk"}
...
storagenode/5 14q49vBKkg 12:47:57.970 | ERROR    agreements    failed to recv response    {"error": "rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
storagenode/0 1p9RqFGtwo 12:47:57.981 | ERROR    agreements    failed to recv response    {"error": "rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
storagenode/7 12RKfd4x7v 12:47:57.988 | ERROR    agreements    failed to recv response    {"error": "rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
...
storagenode/0 1p9RqFGtwo 12:52:41.592 | DEBUG    agreements    is running    {"duration": "1m0s"}
storagenode/0 1p9RqFGtwo 12:52:41.593 | INFO    agreements    sending agreements to satellite    {"number of agreements": 3, "satellite id": "1Z2GQsrLwwWNE8VX5oR6UAH9ZBkAX3nMQ7i4xUsaR3RUKna8Va"}
```

Well done :)

**Pull request merged**